cmd/scaffold.go

```diff
@@ -167,8 +168,22 @@ func (fnb *FlowNodeBuilder) enqueueNetworkInit() {
 			myAddr = fnb.BaseConfig.bindAddr
 		}
 
-		mw, err := p2p.NewMiddleware(fnb.Logger.Level(zerolog.ErrorLevel), codec, myAddr, fnb.Me.NodeID(),
-			fnb.networkKey, fnb.Metrics.Network, p2p.DefaultMaxUnicastMsgSize, p2p.DefaultMaxPubSubMsgSize,
+		// creates a context for middleware
+		// context is used to stop middleware and libp2p
+		ctx, cancel := context.WithCancel(context.Background())
+
+		// initializes middleware
+		mw, err := p2p.NewMiddleware(fnb.Logger.Level(zerolog.ErrorLevel),
+			ctx,
+			cancel,
+			p2p.DefaultLibP2PNodeFactory,
```

Comment:
Passes default libp2p node factory as a dependency to middleware, as was the purpose of this issue.